content-fragment-workflows/src/web-src/src/components/Workflows/ContentFragmentsSelection.js

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {
+  Cell,
+  Checkbox,
+  Column,
+  Heading,
+  Row,
+  TableView,
+  TableBody,
+  TableHeader,
+  Text,
+  View,
+  Flex,
+  ProgressBar,
+  StatusLight,
+} from "@adobe/react-spectrum";
+import { useEffect, useState } from "react";
+import { getContentFragments } from "../utils";
+
+const extractReferences = (contentFragments, contentFragmentPaths) => {
+  const references = [];
+
+  for (const contentFragment of contentFragments) {
+    if (contentFragmentPaths.includes(contentFragment.path) && contentFragment.references.length > 0) {
```

Comment:
Its better to create map/set/object one time on top level (Search by index from map/set/object O(1) time complexity) instead of each time search in array (Search by index from array O(n) time complexity)